api/src/convert/curve_scalar.rs

```diff
@@ -52,7 +52,8 @@ impl TryFrom<&external::CurveScalar> for Scalar {
             .get_data()
             .try_into()
             .map_err(|_| ConversionError::ArrayCastError)?;
-        Ok(Scalar::from_bits(bytes))
+        let maybe_scalar: Option<Scalar> = Scalar::from_canonical_bytes(bytes).into();
```

Comment:
`from_bits` is deprecated so this is safer, see [explanation](https://github.com/mobilecoinfoundation/mobilecoin/pull/3544/files#r1322012410) on why this is expected to result in the same `Scalar` when there isn't an error.